Framework/src/DataDump.cxx

```diff
@@ -0,0 +1,295 @@
+    if (representation == 0) {
+      ImGui::SetColumnWidth(0, 40.0f);
+      ImGui::SetColumnWidth(1, 50.0f);
+      ImGui::SetColumnWidth(2, 50.0f);
+      ImGui::SetColumnWidth(3, 50.0f);
+      ImGui::SetColumnWidth(4, 50.0f);
+    } else if(representation == 1) { // binary
+      ImGui::SetColumnWidth(0, 40.0f);
+      ImGui::SetColumnWidth(1, 243.0f);
+      ImGui::SetColumnWidth(2, 243.0f);
+      ImGui::SetColumnWidth(3, 243.0f);
+      ImGui::SetColumnWidth(4, 243.0f);
+    }
+  }
+  firstDrawColumns = false;
+}
+
+void updatePayloadGui()
+{
+  if (DataDump::guiState.current_payload.data == nullptr) {
+    ImGui::Text("No data loaded yet, click Next.");
+  } else { // all the stuff below should go to a method
+
+    static int representation = 0, old_representation = 1;
+    old_representation = representation;
+    ImGui::RadioButton("hexadecimal", &representation, 0);
+    ImGui::SameLine();
+    ImGui::RadioButton("binary", &representation, 1);
+
+    // scrollable area
+    ImGui::BeginChild("##ScrollingRegion", ImVec2(0, ImGui::GetFontSize() * 25), false);
+    // table
+    ImGui::Columns(5, "payload_display", true);
+
+    // header row
+    ImGui::Separator();
+    resizeColumns(representation, old_representation);
+    ImGui::Text("");
+    ImGui::NextColumn();
+    ImGui::Text("#1");
+    ImGui::NextColumn();
+    ImGui::Text("#2");
+    ImGui::NextColumn();
+    ImGui::Text("#3");
+    ImGui::NextColumn();
+    ImGui::Text("#4");
+    ImGui::NextColumn();
+    ImGui::Separator();
+
+    // print the hex values in the columns and rows of the table
+    vector<string> formattedData =
+      (representation == 0)
+        ? getHexRepresentation(DataDump::guiState.current_payload.data, DataDump::guiState.current_payload.size)
+        : getBinRepresentation(DataDump::guiState.current_payload.data, DataDump::guiState.current_payload.size);
+    int line = 0;
+    static int selected = -1;
+    for (unsigned long pos = 0; pos < formattedData.size();) {
+      char label[32];
+      sprintf(label, "%04d", line * 4);
+      if (ImGui::Selectable(label, selected == line, ImGuiSelectableFlags_SpanAllColumns)) {
+        selected = line;
+      }
+      for (int i = 0; i < 4; i++) { // four columns
+        ImGui::NextColumn();
+        string temp;
+        for (int j = 0; j < 2; j++) { // 2 x 16 bits words (char)
+          if (pos < formattedData.size()) {
+            temp += formattedData[pos];
+            pos++;
+          }
+        }
+        ImGui::TextUnformatted(temp.c_str());
+      }
+      ImGui::NextColumn();
+      line++;
+    }
+
+    // footer
+    ImGui::Columns(1);
+    ImGui::EndChild();
+    ImGui::Separator();
+  }
+}
+
+void updateHeaderGui()
+{
+  if (DataDump::guiState.current_header.data == nullptr) {
+    ImGui::Text("No data loaded yet, click Next.");
+  } else { // all the stuff below should go to a method
+    auto* header = header::get<header::DataHeader*>(DataDump::guiState.current_header.data);
```

Comment:
`get` will return `nullptr` if it does not find `DataHeader`. Could you add `assert(header)` or sth similar?

Reply:
I have added a message in the gui in such a case and return without crashing.